File: objective-c/ArcGISSample/Magnifier/MagnifierViewController.m
//
//  MagnifierViewController.m
//  ArcGISSample
//
//  Created by esrij on 2015/07/27.
//  Copyright (c) 2015年 esrij. All rights reserved.
//

#import "MagnifierViewController.h"


@implementation MagnifierViewController


- (void)viewDidLoad
{
    [super viewDidLoad];
    
    AGSMapView *agsMapView = [[AGSMapView alloc] initWithFrame:self.view.bounds];
    [self.view addSubview:agsMapView];
    
    //タイルマップサービスレイヤーの追加
    NSURL *url = [NSURL URLWithString:@"http://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer"];
    AGSTiledMapServiceLayer *agsTiledMapServiceLayer = [[AGSTiledMapServiceLayer alloc] initWithURL:url];
    [agsMapView addMapLayer:agsTiledMapServiceLayer withName:@"Tiled Layer"];
    
    AGSPoint *point = [AGSPoint pointWithX:15554789.5566484 y:4254781.24130285 spatialReference:[AGSSpatialReference spatialReferenceWithWKID:102100]];
    [agsMapView zoomToScale:100000 withCenterPoint:point animated:YES];

    //マップ上を長押しすると拡大鏡を表示
    agsMapView.showMagnifierOnTapAndHold = YES;
    
}


- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: swift/ArcGISSample/MagnifierViewController.swift
//
//  MagnifierViewController.swift
//  ArcGISSample
//
//  Created by esrij on 2015/07/27.
//  Copyright (c) 2015年 esrij. All rights reserved.
//

import UIKit
import ArcGIS


class MagnifierViewController: UIViewController {

    override func viewDidLoad() {
    
        super.viewDidLoad()
        
        let agsMapView = AGSMapView(frame: view.bounds)
        view.addSubview(agsMapView)
    
        //タイルマップサービスレイヤーの追加
        let url = URL(string: "https://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer")
        let tiledLyr = AGSTiledMapServiceLayer(url:url)
        agsMapView.addMapLayer(tiledLyr, withName:"Tiled Layer")
            
        let point = AGSPoint(x: 15554789.5566484, y: 4254781.24130285, spatialReference:AGSSpatialReference(wkid: 102100))
        agsMapView.zoom(toScale: 100000, withCenter: point, animated: true)

        //マップ上を長押しすると拡大鏡を表示
        agsMapView.showMagnifierOnTapAndHold = true
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    
}
